bazel/external/quiche.BUILD

```diff
@@ -153,11 +153,14 @@ cc_library(
 )
 
 cc_library(
-    name = "quic_platform_port_utils",
+    name = "quic_platform_for_test",
```

Comment:
This change is incompatible with my #6693. Can you add a new library instead?

Reply:
To clarify: QuicThread is test-only, but quic_port_utils.h is used in production so we need to split them, right?  SGTM.

Reply:
quic_port_utils.h is also test-only, the problem is that I'm trying to use smaller libraries in #6693 and this PR is doing the opposite:)

Reply:
Should we just make sure that goes in first then?

Reply:
@jmarantz: Yes, I'd prefer for that to go in first:)  I just assigned it to you:)

Reply:
Split into small libraries. This PR is blocking #6650. I hope to check in this one ASAP.